tools/crmadmin.c

```diff
@@ -394,15 +428,31 @@ admin_msg_callback(const char *buffer, ssize_t length, gpointer userdata)
 
     } else if (validate_crm_message(xml, crm_system_name, admin_uuid, XML_ATTR_RESPONSE) == FALSE) {
         crm_trace("Message was not a CRM response. Discarding.");
+        printf("Validation of response failed\n");
 
-    } else if (DO_HEALTH) {
+    } else if (DO_HEALTH || DO_PACEMAKERD_HEALTH) {
         xmlNode *data = get_message_xml(xml, F_CRM_DATA);
-        const char *state = crm_element_value(data, XML_PING_ATTR_CRMDSTATE);
-
-        printf("Status of %s@%s: %s (%s)\n",
+        const char *state = DO_PACEMAKERD_HEALTH?
+                                crm_element_value(data, XML_PING_ATTR_PACEMAKERDSTATE):
+                                crm_element_value(data, XML_PING_ATTR_CRMDSTATE);
+        const char *host_from = crm_element_value(xml, F_CRM_HOST_FROM);
+        time_t pinged = (time_t) 0;
+        crm_time_t *crm_when = crm_time_new(NULL);
+        char *pinged_buf = NULL;
+
+        crm_element_value_int(data, XML_ATTR_TSTAMP, (int *) &pinged);
```

Comment:
time_t doesn't have to be the same size as int. Its size isn't fixed, so declare pinged as long long and use crm_element_value_ll(). You can always cast long long to time_t. (Even if time_t is smaller, that just means that platform already can't handle Y2038 values, so this doesn't make things any worse.)